modules/API/pages/gsql-endpoints.adoc

```diff
@@ -0,0 +1,6394 @@
+
+=== saml login
+
+`GET /gsql/v1/saml/authnrequest`
+
+This endpoint is used to saml login.
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X GET 'localhost:8123/gsql/v1/saml/authnrequest' 
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----
+{"error":false,"message":"","results":{"ssoBinding":"GET","OIDCRequest":"https://auth.tgcloud-dev.com/authorize?organization=sample_org_id&scope=openid+profile+email&response_type=id_token+token&redirect_uri=http%3A%2F%2F127.0.0.1%3A14240%2Fapi%2Fauth%2Foidc%2Fcallback&nonce=sample_string&client_id=sample_client_id&response_mode=form_post"}}
+----
+--
+====
+
+'''
+
+=== oidc login
+
+`GET /gsql/v1/oidc/authnrequest`
+
+This endpoint is used to oidc login.
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X GET 'localhost:8123/gsql/v1/saml/meta' 
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----
+{"error":false,"message":"","results":{"ssoBinding":"GET","OIDCRequest":"https://auth.tgcloud-dev.com/authorize?organization=sample_org_id&scope=openid+profile+email&response_type=id_token+token&redirect_uri=http%3A%2F%2F127.0.0.1%3A14240%2Fapi%2Fauth%2Foidc%2Fcallback&nonce=sample_string&client_id=sample_client_id&response_mode=form_post"}}
+----
+--
+====
+
+'''
+
+=== create new JWT token
+
+`POST /gsql/v1/tokens`
+
+This endpoint is used to create a new JWT token.
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X POST 'localhost:8123/gsql/v1/tokens'
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----			
+{"expiration":"Wed Jul 10 06:12:47 UTC 2024","error":false,"message":"Generate new JWT token successfully.","token":"eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0aWdlcmdyYXBoIiwiaWF0IjoxNzE5OTg3MTYyLCJleHAiOjE3MjA1OTE5NjcsImlzcyI6IlRpZ2VyR3JhcGgiLCJncmFwaCI6InBvY19ncmFwaCJ9.w4ms-si5egtbhI2Cms7uL1qUU8oG0S08KIiaG_VL3Fs"}
+----
+--
+====
+==== Payload:
+
+//[width="100%",cols="25%,25%,50%",options="header",]
+[%autowidth]
+|===
+| Name | Required | Description
+
+| `secret`
+| no
+| the secret denotes the user 
+
+| `graph`
+| no
+| the graph the token created for
+
+| `lifetime`
+| no
+| the duration time of the token
+|===
+
+'''
+
+=== drop JWT tokens
+
+`DELETE /gsql/v1/tokens?clear=<true/false>`
+
+This endpoint is used to drop specific a list of tokens.
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X DELETE 'localhost:8123/gsql/v1/tokens' -d {"tokens": "$token1,$token2"}
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----			
+{"error":false,"message":"Successfully dropped the specified JWT tokens"}
+----
+--
+====
+==== Parameters:
+
+//[width="100%",cols="25%,25%,50%",options="header",]
+[%autowidth]
+|===
+| Name | Required | Description
+
+| `clear`
+| no
+| default:false. clear is true means clear out the current block list.

```

Comment:
@quanchentg  `true means clear out the current block list, default value is false.`